Sources/App/Commands/Common.swift

```diff
@@ -13,7 +13,12 @@ func updatePackage(application: Application,
                     .load(on: application.db)
                     .flatMap { pkg.$versions.load(on: application.db) }
                     .flatMap {
-                        pkg.status = .ok
+                        if stage == .ingestion && pkg.status == .none {
+                            // newly ingested package: leave status == .none for fast-track
+                            // analysis
+                        } else {
+                            pkg.status = .ok
+                        }
```

Comment:
This is updates the processing order in the simplest way: we keep status a `.none` until a package has passed all processing stages. Our stage selection already prioritises packages with status = `.none`, so this works out nicely and I don't think it's too much of a stretch to interpret status in this way.

The alternative isn't too involved either but would require us to introduce a `.new` state. If we were to do that we should make a schema migration and make `status` non-optional with a default of `.new`. At that point we're essentially simply replacing `.none` (or `NULL`) with `.new` and I'm not really sure that's worth it.

What do you think?

Reply:
One advantage of this change in general, whether we go with `.none` or `.new`, is that we can easily see packages that are pending first full processing. Previously, setting them to `.ok` after ingestion made them indistinguishable from re-ingested packages.

Reply:
This looks good to me, and makes perfect sense.

If I was looking at this without the debt of an existing database field, it'd be a clear vote for `.new` being a default. It's explicit, and obvious, and that's a good thing.

I'd say there's an argument for changing it. It may be super obvious what it all means now as it's fresh in our minds, but will that still be the case when it's a few months in? I don't think we *have* to do it, but I'd say it's worth considering.

Reply:
Yeah, I agree. The `.none` reads weird in the tests, even knowing what it means now. I'll transition that over!